pynamodb/attributes.py

```diff
@@ -372,6 +370,22 @@ def deserialize(self, attribute_values: Dict[str, Dict[str, Any]]) -> None:
                 value = attr.deserialize(attr.get_value(attribute_value))
                 setattr(self, name, value)
 
+    @classmethod
+    def _instantiate(cls: Type[_ACT], attribute_values: Dict[str, Dict[str, Any]]) -> _ACT:
+        discriminator_attr = cls._get_discriminator_attribute()
+        if discriminator_attr:
+            discriminator_attribute_value = attribute_values.pop(discriminator_attr.attr_name, None)
+            if discriminator_attribute_value:
```

Comment:
Why would we allow someone to have a discriminator attribute with no value? (I see this is just being moved from down below, so sorry if this has been answered before)

Reply:
My common use case is for an abstract base class. See the TypedValue class in test_discriminator.py